src/auto-evo/simulation/food_source/EnvironmentalFoodSource.cs

```diff
@@ -42,5 +44,55 @@ public override float TotalEnergyAvailable()
         {
             return totalEnvironmentalEnergySource;
         }
+
+        protected override float StorageScore(MicrobeSpecies species, Compound compound, Patch patch,
+            SimulationCache simulationCache)
+        {
+            if (simulationCache.DayNightCycleEnabled)
+            {
+                var dayNightCycleConfiguration = SimulationParameters.Instance.GetDayNightCycleConfiguration();
+                var nightTime = dayNightCycleConfiguration.DaytimePercentage * dayNightCycleConfiguration.HoursPerDay;
+
+                // If a species consumes a lot, it ought to store more.
+                // NB: this might artificially penalize overproducers but I'm willing to accept it for now - Maxonovien
+                var reserveScore = CompoundUse(species, compound, patch, simulationCache) *
+                    simulationCache.GetStorageCapacityForSpecies(species);
+
+                // Severely penalize species not storing enough for their production
+                var minimumViableReserveScore = Constants.AUTO_EVO_MINIMUM_VIABLE_RESERVE_PER_TIME_UNIT * nightTime;
+
+                if (reserveScore <= minimumViableReserveScore)
+                    return minimumViableReserveScore / Constants.AUTO_EVO_NON_VIABLE_RESERVE_PENALTY;
+
+                // TODO: consider using a cap, e.g two times the viable reserve (i.e. osmoregulation + base movement.
+                return reserveScore;
+            }
+
+            return 1.0f;
+        }
+
+        // TODO CACHE IT, AS FOR ENERGY GENERATION, TO AVOID MULTIPLE LOOPS.
```

Comment:
As the TODO says this should be cached. This PR already adds caching for the storage count, so adding one more cached thing should be relatively easy.